editor/import/scene_import_settings.cpp

```diff
@@ -119,14 +125,12 @@ void SceneImportSettings::_fill_material(Tree *p_tree, const Ref<Material> &p_ma
 	Ref<Texture2D> icon = get_theme_icon(SNAME("StandardMaterial3D"), SNAME("EditorIcons"));
 
 	TreeItem *item = p_tree->create_item(p_parent);
-	item->set_text(0, p_material->get_name());
-	item->set_icon(0, icon);
-
-	bool created = false;
-	if (!material_set.has(p_material)) {
-		material_set.insert(p_material);
-		created = true;
+	if (p_material->get_name().is_empty()) {
+		item->set_text(0, TTR("<Unnamed Material>"));
```

Comment:
@Calinou Reviews welcome.

I would use the classname of the p_material with deduplication.

Reply:
I agree displaying a material ID somehow would be useful. I'd suggest using the form `[Unnamed Material 1234]`). Is there a prefereably deterministic material ID coming from the import system?

Reply:
For what I understand from the import UI, the scene importer can provide a material ID.
But, if the ID is not provider, an autogenerated ID with a progressive number is created [here](https://github.com/godotengine/godot/pull/59834/files#diff-26e1bde2ef7dbe454265bd0a7abc7fea36035cc84786acfd3bce9ee40edc8228R110).
I can display that (or a formatted version), but it's not deterministic.

Reply:
Having a non deterministic name is worse than an empty string.

Reply:
Different icons can be used here to distinguish whether the name is a deterministic material ID coming from the import system.